fluent/toInterfaceValue.go

```diff
@@ -0,0 +1,65 @@
+package fluent
+
+import (
+	"errors"
+
+	"github.com/ipld/go-ipld-prime/datamodel"
+)
+
+var ErrInvalidKind = errors.New("invalid kind")
+var ErrUnknownKind = errors.New("unknown kind")
```

Comment:
nit: I would avoid exposing these errors in the API for now, because once they are exposed, they are part of the API we must support. These errors should be very rare and almost always due to a bug that needs to be fixed in someone's code, so it also doesn't seem like errors that the end user should be checking against.